### DISCUSSION TITLE
Real-world open source project based on TypeScript, DDD, ES & CQRS practices

### DISCUSSION THREAD

**Discussion by marcus-sa** [techexcellenceio/techexcellence#17] [Ideas]
Is anyone interested in such a project? I'm currently building https://github.com/convoy-js/convoy (which is a continuation of https://github.com/convoy-js/nest-convoy/tree/feat/mikro-orm).

It's a WIP full fledged framework based around these topics for building distributed event driven systems:
- DDD
- CQRS
- Event Sourcing
- and more

If the interest is there, I could finish the framework.
Inspiration was mainly taking from https://eventuate.io/

**Comment by valentinajemuovic** [comment]
Thanks @marcus-sa this is a good initiative. In terms of JS, I know there is a lot of demand, esp. since most examples on the internet for these topics are mainly Java, so it would be great to have something in JS/TS.

The topics you described above is also similar to my future intentions for Java/.NET, having an example for distributed systems.

I sent you a DM to discuss some possibilities.
